extensions/standard-processors/processors/RetryFlowFile.cpp

```diff
@@ -0,0 +1,212 @@
+      logger_->log_error("Narrowing Exception: %s", e.what());
+      session->transfer(flow_file, Failure);
+      return;
+    }
+    if (penalize_retries_) {
+      session->penalize(flow_file);
+    }
+    session->transfer(flow_file, Retry);
+    return;
+  }
+  if (!setRetriesExceededAttributesOnFlowFile(context, flow_file)) {
+    session->transfer(flow_file, Failure);
+    yield();
+    return;
+  }
+  session->transfer(flow_file, RetriesExceeded);
+}
+
+void RetryFlowFile::readDynamicPropertyKeys(core::ProcessContext* context) {
+  exceeded_flowfile_attribute_keys.clear();
+  const std::vector<std::string> dynamic_prop_keys = context->getDynamicPropertyKeys();
+  logger_->log_info("RetryFlowFile registering %d keys", dynamic_prop_keys.size());
+  for (const auto& key : dynamic_prop_keys) {
+    exceeded_flowfile_attribute_keys.emplace_back(core::PropertyBuilder::createProperty(key)->withDescription("auto generated")->supportsExpressionLanguage(true)->build());
+    logger_->log_info("RetryFlowFile registered attribute '%s'", key);
+  }
+}
+
+// Returns (1, true) on non-numerical or out-of-bounds retry value
+std::pair<uint64_t, bool> RetryFlowFile::getRetryPropertyValue(const std::shared_ptr<FlowFileRecord>& flow_file) {
+  std::string value_as_string;
+  try {
+    if (flow_file->getAttribute(retry_attribute_, value_as_string)) {
+      return std::make_pair(std::stoul(value_as_string), false);
+    }
+  }
+  catch(const std::invalid_argument&) {
+    if (fail_on_non_numerical_overwrite_) {
+      logger_->log_info("Non-numerical retry property in RetryFlowFile. Sending flowfile to failure...", value_as_string);
+      return std::make_pair(1, true);
+    }
+    logger_->log_info("Non-numerical retry property in RetryFlowFile: overwriting %s with 1.", value_as_string);
+  }
+  catch(const std::out_of_range&) {
+    logger_->log_error("Narrowing Exception for %s, treating it as non-numerical value", value_as_string);
+  }
+  return std::make_pair(1, false);
+}
+
+// Returns true on fail on reuse scenario
+bool RetryFlowFile::updateUUIDMarkerAndCheckFailOnReuse(const std::shared_ptr<FlowFileRecord>& flow_file) {
+  const std::string last_retried_by_property_name = retry_attribute_ + ".uuid";
+  const std::string current_processor_uuid = getUUIDStr();
+  std::string last_retried_by_uuid;
+  if (flow_file->getAttribute(last_retried_by_property_name, last_retried_by_uuid)) {
+    if (last_retried_by_uuid != current_processor_uuid) {
+      if (reuse_mode_ == FAIL_ON_REUSE) {
+        logger_->log_error("FlowFile %s was previously retried with the same attribute by a different "
+            "processor (uuid: %s, current uuid: %s). Transfering flowfile to 'failure'...",
+            flow_file->getUUIDStr(), last_retried_by_uuid, current_processor_uuid);
+        return true;
+      }
+      if (reuse_mode_ == WARN_ON_REUSE) {
+        logger_->log_warn("Reusing retry attribute that belongs to different processor. Resetting value to 1.");
```

Comment:
where is it actually reset to `1`?

Reply:
😱 it is not. Will add a reset to 0 instead of 1 on the flowfile attribute here as well.